traffic_ops/app/db/migrations/20200331031637_add_deliveryservice_topology.sql

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+* License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+ALTER TABLE deliveryservice
+    ADD COLUMN topology text,
+    ADD CONSTRAINT deliveryservice_topology_fkey FOREIGN KEY (topology) REFERENCES topology (name) ON UPDATE CASCADE ON DELETE CASCADE;
```

Comment:
also, it's generally not required to explicitly name constraints like this and define indexes for foreign keys. Postgres will create those indexes and name them automatically as long as you do something like this for instance:
```
ALTER TABLE deliveryservice ADD COLUMN topology text REFERENCES topology(name) ON UPDATE CASCADE ON DELETE RESTRICT;
```
And that reminds me, this should be `ON DELETE RESTRICT` because we don't want to allow a Topology to be deleted if a delivery service is still assigned, right?

Reply:
I'm +1 on explicitly naming things like this.

We've had cases in the past where automatically named constraints, keys, etc ended up with different names on different installations, and then we later needed to make a migration to modify the symbol, and it was difficult or impossible, because the name was different on different systems.

Reply:
I'm not -1 on explicit names -- I would be fine either way, I'm just saying it's generally not required.